src/ScalarMessaging.ts

```diff
@@ -841,7 +841,7 @@ async function readEvents(
     }
 }
 
-const onMessage = function (event: MessageEvent<any>): void {
+const onMessage = async function (event: MessageEvent<any>): Promise<void> {
```

Comment:
Are we sure this method can be async? It could make events be handled out-of-order unless we add something to serialise the data flow

Reply:
Good catch, we already have multiple asynchronous call in this function. Removing the async.